### ISSUE TITLE
nestedtensor inconsistency in `torch.masked_select`

### ISSUE THREAD

**Issue by rustamzh** [pytorch/pytorch#168253]
### 🐛 Describe the bug

Here is the code that left me with questions: I am not sure if it is a bug, but I feel it is not it would be a great addition to the docs. I would expect padded nt and padded  nt1 to have the same values at the end of the script, but they are not. If it is not a bug, how can I achieve it: create a nested tensor from a padded tensor and a mask that will have a proper max_len?

```python
import torch

lengths = [5,5,6,6,6,7,7,7,7,8,8,8,8,9]
results = []
for length in lengths:
    results.append(torch.ones((length,)))

results
# [tensor([1., 1., 1., 1., 1.]),
#  tensor([1., 1., 1., 1., 1.]),
#  tensor([1., 1., 1., 1., 1., 1.]),
#  tensor([1., 1., 1., 1., 1., 1.]),
#  tensor([1., 1., 1., 1., 1., 1.]),
#  tensor([1., 1., 1., 1., 1., 1., 1.]),
#  tensor([1., 1., 1., 1., 1., 1., 1.]),
#  tensor([1., 1., 1., 1., 1., 1., 1.]),
#  tensor([1., 1., 1., 1., 1., 1., 1.]),
#  tensor([1., 1., 1., 1., 1., 1., 1., 1.]),
#  tensor([1., 1., 1., 1., 1., 1., 1., 1.]),
#  tensor([1., 1., 1., 1., 1., 1., 1., 1.]),
#  tensor([1., 1., 1., 1., 1., 1., 1., 1.]),
#  tensor([1., 1., 1., 1., 1., 1., 1., 1., 1.])]


nt = torch.nested.nested_tensor(results, layout=torch.jagged)

nt
# NestedTensor(size=(14, j1), offsets=tensor([ 0,  5, 10, 16, 22, 28, 35, 42, 49, 56, 64, 72, 80, 88, 97]), contiguous=True)

pt_infer = torch.nested.to_padded_tensor(nt, 0.0)

pt_infer.shape
# torch.Size([14, 9])

mask = pt_infer != 0

mask.shape
# torch.Size([14, 9])

nt1 = torch.nested.masked_select(pt_infer, mask)

nt1.shape
# torch.Size([14, j2])

nt.shape
# torch.Size([14, j1])

nt1.to_padded_tensor(0.0, ).shape
# torch.Size([14, 97])

torch.nested.to_padded_tensor(nt1, 0.0).shape
# torch.Size([14, 97])

torch.nested.to_padded_tensor(nt, 0.0).shape
# torch.Size([14, 9])

```

### Versions

PyTorch version: 2.9.0+cu128
Is debug build: False
CUDA used to build PyTorch: 12.8
ROCM used to build PyTorch: N/A

OS: Rocky Linux 9.4 (Blue Onyx) (x86_64)
GCC version: (GCC) 11.4.1 20231218 (Red Hat 11.4.1-3)
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.34

Python version: 3.12.12 | packaged by conda-forge | (main, Oct 22 2025, 23:25:55) [GCC 14.3.0] (64-bit runtime)
Python platform: Linux-5.14.0-427.13.1.el9_4.x86_64-x86_64-with-glibc2.34
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to:
GPU models and configuration: GPU 0: NVIDIA H100 80GB HBM3
Nvidia driver version: 575.57.08
cuDNN version: Could not collect
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==2.3.4
[pip3] nvidia-cublas-cu12==12.8.4.1
[pip3] nvidia-cuda-cupti-cu12==12.8.90
[pip3] nvidia-cuda-nvrtc-cu12==12.8.93
[pip3] nvidia-cuda-runtime-cu12==12.8.90
[pip3] nvidia-cudnn-cu12==9.10.2.21
[pip3] nvidia-cufft-cu12==11.3.3.83
[pip3] nvidia-curand-cu12==10.3.9.90
[pip3] nvidia-cusolver-cu12==11.7.3.90
[pip3] nvidia-cusparse-cu12==12.5.8.93
[pip3] nvidia-cusparselt-cu12==0.7.1
[pip3] nvidia-nccl-cu12==2.27.5
[pip3] nvidia-nvjitlink-cu12==12.8.93
[pip3] nvidia-nvtx-cu12==12.8.90
[pip3] pytorch-lightning==2.5.6
[pip3] torch==2.9.0
[pip3] torch-dct==0.1.6
[pip3] torchaudio==2.9.0
[pip3] torchmetrics==1.8.2
[pip3] triton==3.5.0
[conda] numpy                     2.3.4                    pypi_0    pypi
[conda] nvidia-cublas-cu12        12.8.4.1                 pypi_0    pypi
[conda] nvidia-cuda-cupti-cu12    12.8.90                  pypi_0    pypi
[conda] nvidia-cuda-nvrtc-cu12    12.8.93                  pypi_0    pypi
[conda] nvidia-cuda-runtime-cu12  12.8.90                  pypi_0    pypi
[conda] nvidia-cudnn-cu12         9.10.2.21                pypi_0    pypi
[conda] nvidia-cufft-cu12         11.3.3.83                pypi_0    pypi
[conda] nvidia-curand-cu12        10.3.9.90                pypi_0    pypi
[conda] nvidia-cusolver-cu12      11.7.3.90                pypi_0    pypi
[conda] nvidia-cusparse-cu12      12.5.8.93                pypi_0    pypi
[conda] nvidia-cusparselt-cu12    0.7.1                    pypi_0    pypi
[conda] nvidia-nccl-cu12          2.27.5                   pypi_0    pypi
[conda] nvidia-nvjitlink-cu12     12.8.93                  pypi_0    pypi
[conda] nvidia-nvtx-cu12          12.8.90                  pypi_0    pypi
[conda] pytorch-lightning         2.5.6                    pypi_0    pypi
[conda] torch                     2.9.0                    pypi_0    pypi
[conda] torch-dct                 0.1.6                    pypi_0    pypi
[conda] torchaudio                2.9.0                    pypi_0    pypi
[conda] torchmetrics              1.8.2                    pypi_0    pypi
[conda] triton                    3.5.0                    pypi_0    pypi

cc @cpuhrsch @jbschlosser @bhosmer @drisspg @soulitzer @davidberard98 @YuqingJ
